sympy/ntheory/residue_ntheory.py

```diff
@@ -773,6 +777,8 @@ def nthroot_mod(a, n, p, all_roots=False):
     # see Hackman "Elementary Number Theory" (2009), page 76
     if not is_nthpow_residue(a, n, p):
         return None
+    if isprime(p) is False:
```

Comment:
It seems that `isprime()` will return `True` or `False`. Hence this could be `if not isprime(p):`.